pkg/extensions/sync/destination.go

```diff
@@ -254,6 +387,25 @@ func (registry *DestinationRegistry) copyManifest(repo string, desc ispec.Descri
 			}
 		}
 
+		// If we've filtered the manifest list, we need to update it
+		if registry.config != nil &&
+			len(registry.config.Platforms) > 0 &&
+			len(filteredManifests) != len(indexManifest.Manifests) && len(filteredManifests) > 0 {
+			// Create a new index with the filtered manifests
+			indexManifest.Manifests = filteredManifests
+
+			// Update the manifest content with the filtered list
+			updatedContent, err := json.Marshal(indexManifest)
+			if err != nil {
+				registry.log.Error().Str("errorType", common.TypeOf(err)).
+					Err(err).Str("repository", repo).
+					Msg("failed to marshal updated index manifest")
+				return err
+			}
+
+			manifestContent = updatedContent
```

Comment:
Note this will break any referrer pointing to the original index digest.
Unless the same digest is kept in the filename, but a different one would be computed based on the new filename, are we doing this?